builder/builder-source-archive.c

```diff
@@ -409,6 +410,22 @@ unzip (GFile   *dir,
   return res;
 }
 
+static gboolean
+unrpm (GFile   *dir,
+       const char *rpm_path,
+       GError **error)
+{
+  gboolean res;
+  const gchar *argv[] = { "sh", "-c", NULL, NULL };
+  char *unrpm_cmdline = g_strdup_printf("rpm2cpio %s | cpio -i -d", rpm_path);
```

Comment:
It's probably a good idea, especially given Matthias's suggestions below. It can't be worse that this ugly 'sh -c "rpm2cpio | cpio -id" ' invoked from C code if you ask me ;) No idea how acceptable it is to add such dependencies to flatpak though.